liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/BasicSupport.java

```diff
@@ -410,6 +410,33 @@ protected void installLicense() throws MojoExecutionException, IOException {
         }
     }
     
+    // delete application file from server apps or dropins directory, including loose application
+    protected void deleteApplication(File artifact) throws IOException {
+        deleteApplication(new File(serverDirectory, "apps"), artifact.getName());
+        if (artifact.getName().endsWith(".xml")) {
+            deleteApplication(new File(serverDirectory, "apps"), artifact.getName().substring(0, artifact.getName().length() - 4));
+        } else {
+            deleteApplication(new File(serverDirectory, "apps"), artifact.getName() + ".xml");
+        }
+        
+        deleteApplication(new File(serverDirectory, "dropins"), artifact.getName());
+        if (artifact.getName().endsWith(".xml")) {
+            deleteApplication(new File(serverDirectory, "dropins"), artifact.getName().substring(0, artifact.getName().length() - 4));
+        } else {
+            deleteApplication(new File(serverDirectory, "dropins"), artifact.getName() + ".xml");
+        }
+    }
+    
+    protected void deleteApplication(File parent, String filename) throws IOException {
+        File application = new File(parent, filename);
+        if (application.isDirectory()) {
```

Comment:
Our plugin doesn't support expanded option for install-app yet but we could add the expanded war support in the future.
https://developer.ibm.com/wasdev/docs/article_appdeployment/

Also the application can also be expanded by the server runtime `<applicationManager autoExpand="true"/>` , which I have missed.